controllers/replicationdestination_controller.go

```diff
@@ -93,36 +103,40 @@ func (r *ReplicationDestinationReconciler) Reconcile(ctx context.Context, req ct
 		return result, err
 	}
 
-	if r.countReplicationMethods(inst, logger) > 1 {
-		err = fmt.Errorf("only a single replication method can be provided")
-		return result, err
-	}
-	result, err = reconcileDestUsingCatalog(ctx, inst, r, logger)
-	if errors.Is(err, errNoMoverFound) { // do the old stuff
-		if inst.Spec.External != nil {
-			// Not an internal method... we're done.
-			return ctrl.Result{}, nil
-		}
-		err = fmt.Errorf("a replication method must be specified")
-	}
+	rdm, err := newRDMachine(inst, r.Client, logger, record.NewEventRecorderAdapter(r.EventRecorder))
 
-	// Set reconcile status condition
-	if err == nil {
+	// Using only external method
+	if errors.Is(err, mover.ErrNoMoverFound) && inst.Spec.External != nil {
+		return ctrl.Result{}, nil
+	}
+	// Both internal and external methods defined
+	if rdm != nil && inst.Spec.External != nil {
+		err = mover.ErrMultipleMoversFound
 		apimeta.SetStatusCondition(&inst.Status.Conditions, metav1.Condition{
-			Type:    volsyncv1alpha1.ConditionReconciled,
-			Status:  metav1.ConditionTrue,
-			Reason:  volsyncv1alpha1.ReconciledReasonComplete,
-			Message: "Reconcile complete",
+			Type:    volsyncv1alpha1.ConditionSynchronizing,
+			Status:  metav1.ConditionFalse,
+			Reason:  volsyncv1alpha1.SynchronizingReasonError,
+			Message: err.Error(),
 		})
-	} else {
+	}
+	// No method found
+	if rdm == nil && inst.Spec.External == nil {
+		err = mover.ErrNoMoverFound
 		apimeta.SetStatusCondition(&inst.Status.Conditions, metav1.Condition{
-			Type:    volsyncv1alpha1.ConditionReconciled,
+			Type:    volsyncv1alpha1.ConditionSynchronizing,
 			Status:  metav1.ConditionFalse,
-			Reason:  volsyncv1alpha1.ReconciledReasonError,
+			Reason:  volsyncv1alpha1.SynchronizingReasonError,
 			Message: err.Error(),
 		})
 	}
 
+	// All good, so run the state machine
+	if err == nil {
+		result, err = sm.Run(ctx, rdm, logger)
+		// Retrieve any Status info modified by the state machine
+		inst.Status = rdm.ReplicationDestination.Status
```

Comment:
Is there a reason we need a copy of the ReplicationDestination in the rdMachine?  Then we wouldn't need this extra step to always copy over the status at the end

Reply:
Maybe not... Let me take a look at switching to a *RD instead of embedding it. I embedded it originally so that the methods would be exposed, but they may not need to be.

Reply:
done